src/libsystemd-network/network-internal.c

```diff
@@ -255,11 +255,10 @@ int config_parse_ifalias(const char *unit,
                 return 0;
         }
 
-        free(*s);
-        if (*n)
-                *s = TAKE_PTR(n);
+        if (isempty(n))
+                *s = mfree(*s);
         else
-                *s = NULL;
+                free_and_replace(*s, n);
```

Comment:
This block looks strange because `free_and_replace` is used everywhere else. Is there something special about `ifalias` or is the handling of an empty string not correct in the other places?